enhancements/subscription-content/subscription-injection.md

```diff
@@ -0,0 +1,397 @@
+(cri-o, podman, buildah, etc.).
+
+In OpenShift 4 the preferred operating system (RHCOS) is not capable of attaching subscriptions
+individually. To access subscription content, containers must be provided entitlement keys and
+`redhat.repo` configurations via other means. Enabling this capability needs to be simple and
+straightforward.
+
+### Goals
+
+1. Inject entitlement keys and `redhat.repo` configurations into a pod spec.
+2. Define APIs and conventions that allow cluster admins to add the required information to a
+   cluster (entitlement keys and `redhat.repo` configurations).
+
+### Non-Goals
+
+1. Automatically install entitlement keys and `redhat.repo` configurations onto a subscribed
+   cluster.
+
+## Proposal
+
+### User Stories
+
+#### Add debug tools to containers
+
+As a developer debugging my application I want access to RHEL subscription content in my containers
+So that I can install debugging tools from Red Hat via `yum install`
+
+#### Access appropriate RHEL content for container
+
+As a developer using OpenShift I want to be able to access appropriate RHEL content for my container
+So that I download RHEL 8 content in UBI 8 containers And download RHEL 7 content in UBI 7
+containers
+
+### Implementation Details/Notes/Constraints [optional]
+
+#### Installation and Setup
+
+When the `subscription-injection-operator` is installed, it creates a subscription `ClusterBundle`
+custom resource definition. The operator deployment creates two default instances, which are
+cluster-scoped:
+
+```yaml
+kind: ClusterBundle
+apiVersion: subscription.openshift.io/v1
+metadata:
+  name: rhel7
+spec:
+  entitlement:
+    name: etc-pki-entitlement
+  yumRepository:
+    name: redhat-repo-rhel7
+```
+
+and
+
+```yaml
+kind: ClusterBundle
+apiVersion: subscription.openshift.io/v1
+metadata:
+  name: rhel8
+spec:
+  entitlement:
+    name: etc-pki-entitlement
+  yumRepository:
+    name: redhat-repo-rhel8
+```
+
+The `entitlement` object is a reference to a `Secret` in the `openshift-config` namespace containing
+the entitlement keys for the cluster's subscription. The `yumRepository` object is a reference to a
+`ConfigMap` in the `openshift-config` namespace containing the yum repository definitions for the
+desired major version of RHEL.
+
+The cluster administrator is responsible for obtaining the `entitlement` and `yumRepository`
+information and adding them to the cluster. Administrators can refer to existing guidance on how to
+obtain this information.
+
+#### Subscription Bundle Controller
+
+When a `ClusterBundle` object is created, the Subscription Bundle Controller (a new controller) will
+perform the necessary actions to make the bundle available to cluster workloads. This consists of
+the following:
+
+1. Verify that the referenced `entitlement` and `yumRepository` resources exist in the
+   `openshift-config` namespace.
+2. Create a share for the `entitlement` and `yumRepository` resources as declared in the Projected
```

Comment:
Can we get a little more precise here ... "create ProjectedResources instances per the CSI driver proposal for the entitlement and yumRepository resources

Reply:
will update in the next push.